### PR TITLE
remove test sample verbiage from datasets

### PR DESCRIPTION
**Pull request by sandalns**
### Please check if the PR fulfills these requirements

- [ ] Include reference to internal ticket and/or GitHub issue "Fixes #NNNN" (if applicable)
- [ ] Relevant tests for the changes have been added
- [ ] Relevant docs have been added / updated

Fixes: KOL-3519

### CODE REVIEW

**Comment by ly-kolena** [comment]
What brought this up? The use of `TEST_SAMPLE` is for compatibility with current WebUI, are we planning to diverge?

File: kolena/_experimental/dataset/_dataset.py [hunk 1]
```diff
@@ -36,25 +36,25 @@
...
 
-class TestSampleType(str, Enum):
-    CUSTOM = "TEST_SAMPLE/CUSTOM"
-    DOCUMENT = "TEST_SAMPLE/DOCUMENT"
-    IMAGE = "TEST_SAMPLE/IMAGE"
-    POINT_CLOUD = "TEST_SAMPLE/POINT_CLOUD"
-    TEXT = "TEST_SAMPLE/TEXT"
-    VIDEO = "TEST_SAMPLE/VIDEO"
+class DataPointType(str, Enum):
+    CUSTOM = "DATA_POINT/CUSTOM"
```
Comment (ly-kolena) [inline kolena/_experimental/dataset/_dataset.py]: I think it should be `DatapointType` and `DATAPOINT`

File: kolena/_experimental/dataset/_dataset.py [hunk 2]
```diff
@@ -36,25 +36,25 @@
 from kolena.workflow._datatypes import TypedDataObject
 
 COL_DATAPOINT = "datapoint"
-TEST_SAMPLE_TYPE = "TEST_SAMPLE"
 FIELD_LOCATOR = "locator"
 FIELD_TEXT = "text"
+DATAPOINT_TYPE = "DATA_POINT"
```
Comment (ly-kolena) [inline kolena/_experimental/dataset/_dataset.py]: Is this used?

Comment (sandalns) [reply kolena/_experimental/dataset/_dataset.py]: removed

File: kolena/_experimental/dataset/_dataset.py (hunk 1 above)
Comment (sandalns) [reply kolena/_experimental/dataset/_dataset.py]: done

File: kolena/_experimental/dataset/__init__.py [hunk 3]
```diff
@@ -13,8 +13,59 @@
 # limitations under the License.
 from kolena._experimental.dataset._dataset import fetch_dataset
 from kolena._experimental.dataset._dataset import register_dataset
+from kolena.workflow.annotation import BitmapMask
```
Comment (ly-kolena) [inline kolena/_experimental/dataset/__init__.py]: I'm more inclined to relocate these datatypes into a new module, and not tied to `dataset` (as learned from `workflow`)

File: kolena/_experimental/dataset/__init__.py [hunk 4]
```diff
@@ -13,8 +13,59 @@
...
+from kolena.workflow.annotation import ScoredLabeledBoundingBox3D
+from kolena.workflow.annotation import ScoredLabeledPolygon
+from kolena.workflow.annotation import ScoredPolygon
+from kolena.workflow.annotation import SegmentationMask
+from kolena.workflow.asset import Asset
+from kolena.workflow.asset import BinaryAsset
+from kolena.workflow.asset import ImageAsset
+from kolena.workflow.asset import PlainTextAsset
+from kolena.workflow.asset import PointCloudAsset
+from kolena.workflow.ground_truth import GroundTruth
```
Comment (ly-kolena) [inline kolena/_experimental/dataset/__init__.py]: GroundTruth, Inference, and Model should not be included. Classes in `plot` is also not relevant to `dataset`.

Comment (sandalns) [reply kolena/_experimental/dataset/__init__.py]: If we want to offer a new path to import all of these, what do you propose? I'm open to pausing this part of the PR ( just rename TEST_SAMPLE to DATAPOINT), and handling the other imports in a follow up. Should these also be relocated to another module or do we want `kolena.workflow.plot` to still be used?

 @yifanwu-kolena @dylangrandmont  What are your thoughts here?

Comment (yifanwu-kolena) [reply kolena/_experimental/dataset/__init__.py]: just renaming TEST_SAMPLE to DATAPOINT sounds good to me.

i will not worry the importing paths until dataset is in a good stable shape

Comment (sandalns) [reply kolena/_experimental/dataset/__init__.py]: done

**Review by yifanwu-kolena — APPROVED** [review]

**Review by munkyshi — APPROVED** [review]

**Pull request merged**